corgidrp/l2a_to_l2b.py

```diff
@@ -157,27 +157,39 @@ def flat_division(input_dataset, master_flat):
 
     return None
 
-def correct_bad_pixels(input_dataset):
+def desmear(input_dataset):
     """
-    
-    Compute bad pixel map and correct for bad pixels. 
-
-    MMB Notes: 
-        - We'll likely want to be able to accept an external bad pixel map, either 
-        from the CalDB or input by a user. 
-        - We may want to accept just a list of bad pixels from a user too, thus 
-        saving them the trouble of actually making their own map. 
-        - We may want flags to decide which pixels in the DQ we correct. We may 
-        not want to correct everything in the DQ extension
-        - Different bad pixels in the DQ may be corrected differently.
 
+    Desmear data frames.
 
     Args:
         input_dataset (corgidrp.data.Dataset): a dataset of Images (L2a-level)
 
     Returns:
-        corgidrp.data.Dataset: a version of the input dataset with bad pixels corrected
+        corgidrp.data.Dataset: a version of the input dataset with desmear applied
+
     """
 
-    return None
+    data = input_dataset.copy()
+    data_cube = data.all_data
+
+    rowreadtime_sec = detector.get_rowreadtime_sec()
+
+    for i in range(data_cube.shape[0]):
+        exptime_sec = float(data[i].ext_hdr['EXPTIME'])
+        smear = np.zeros_like(data_cube[i])
+        m = len(smear)
+        for r in range(m):
+            columnsum = 0
+            for s in range(r+1):
+                columnsum = columnsum + rowreadtime_sec/exptime_sec*((1 
+                + rowreadtime_sec/exptime_sec)**((s+1)-(r+1)-1))*data_cube[i,s,:]
+            smear[r,:] = columnsum
+        data_cube[i] -= smear
```

Comment:
Given that this is an analytical function, we should be able to calculate the added error that this operation applies to the data. Could please make an issue about this as a future to-do item so we can move this PR forward, but still keep track of the fact that it's not here now.

Reply:
https://github.com/roman-corgi/corgidrp/issues/102